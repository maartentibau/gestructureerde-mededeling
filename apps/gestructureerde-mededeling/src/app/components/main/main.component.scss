@import '../../../assets/styles/screen-media-queries';

.shortcuts {
  margin: 2rem 0.5rem;
  display: flex;
  justify-content: space-around;
  color: #fff;

  a {
    flex: 0 0 auto;
    text-transform: uppercase;
    background-color: #15863c;
    margin: 0 0.5em;
    height: 80px;
    line-height: 80px;
    width: 80px;
    text-align: center;
    display: block;

    &.mat-raised-button > ::ng-deep span.mat-button-wrapper {
      width: 100%;

      & > fa-icon {
        margin-right: 8px;
      }
    }

    &.mat-raised-button {
      color: #fff;
    }

    &:hover {
      color: #216a38;
      background-color: white;
    }

    fa-icon {
      font-size: 2.5rem;
    }
  }

  &.large {
    a {
      font-size: 1.4rem;
      height: 60px;
      flex: 1 1 0;
      line-height: 60px;
      max-width: 450px;
      margin: 0 16px;

      text-overflow: ellipsis;
      white-space: nowrap;
      overflow: hidden;

      fa-icon {
        font-size: 1.5rem;
        line-height: 1.5rem;
      }

      @media #{$mat-md} {
        max-width: 350px;
        height: 50px;
        font-size: 1rem;
        line-height: 50px;

        fa-icon {
          font-size: 1.3rem;
          line-height: 1.5rem;
        }
      }
    }
  }
}
