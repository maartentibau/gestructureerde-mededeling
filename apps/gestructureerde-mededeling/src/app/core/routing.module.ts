import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { CreateComponent } from '../components/create/create.component';
import { GenerateComponent } from '../components/generate/generate.component';
import { ValidateComponent } from '../components/validate/validate.component';

const routes: Routes = [
  { path: 'create', component: CreateComponent },
  { path: 'generate', component: GenerateComponent },
  { path: 'validate', component: ValidateComponent },
  { path: '', redirectTo: '/generate', pathMatch: 'full' },
  { path: '**', component: GenerateComponent },
];

@NgModule({
  imports: [RouterModule.forRoot(routes, { relativeLinkResolution: 'legacy' })],
  exports: [RouterModule],
})
export class RoutingModule {}
