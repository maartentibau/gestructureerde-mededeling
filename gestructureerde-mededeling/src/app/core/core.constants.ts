export const DEFAULT_TITLE = `Gestructureerde Mededeling - OGM`;
