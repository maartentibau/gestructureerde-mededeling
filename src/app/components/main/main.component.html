@if (isSmallScreen$ | async) {
  <div class="shortcuts">
    <a mat-fab disableRipple color="primary" [routerLink]="['/create']" title="Maak uw eigen gestructureerde mededeling"
      ><fa-icon [icon]="['fas', 'star']"></fa-icon
    ></a>

    <a
      mat-fab
      disableRipple
      color="primary"
      [routerLink]="['/generate']"
      title="Genereer een willekeurige gestructureerde mededeling"
      ><fa-icon [icon]="['fas', 'cog']"></fa-icon
    ></a>

    <a
      mat-fab
      disableRipple
      color="primary"
      [routerLink]="['/validate']"
      title="Controleer een gestructureerde mededeling"
      ><fa-icon [icon]="['fas', 'check-circle']"></fa-icon
    ></a>
  </div>
} @else {
  <div class="shortcuts large">
    <a
      mat-raised-button
      disableRipple
      color="primary"
      [routerLink]="['/create']"
      title="Maak uw eigen gestructureerde mededeling"
      ><fa-icon [icon]="['fas', 'star']"></fa-icon>Zelf Mededeling Maken</a
    >

    <a
      mat-raised-button
      disableRipple
      [routerLink]="['/generate']"
      title="Genereer een willekeurige gestructureerde mededeling"
      ><fa-icon [icon]="['fas', 'cog']"></fa-icon>Genereer Mededeling</a
    >

    <a mat-raised-button disableRipple [routerLink]="['/validate']" title="Controleer een gestructureerde mededeling"
      ><fa-icon [icon]="['fas', 'check-circle']"></fa-icon>Controleer Mededeling</a
    >
  </div>
}
