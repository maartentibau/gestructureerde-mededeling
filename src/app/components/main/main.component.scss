@import '../../../assets/styles/screen-media-queries';

.shortcuts {
  margin: 2rem 0.5rem;
  display: flex;
  justify-content: space-around;
  color: #fff;

  a {
    flex: 0 0 auto;
    text-transform: uppercase;
    background-color: #15863c;
    margin: 0 0.5em;
    height: 80px;
    line-height: 80px;
    width: 80px;
    text-align: center;
    display: block;

    &.mdc-button--raised > ::ng-deep span.mdc-button__label {
      display: grid;
      grid-template-columns: max-content auto;
      align-items: center;
      width: 100%;

      & > fa-icon {
        margin-right: 8px;
      }
    }

    &.mdc-button--raised {
      color: #fff;
    }

    &:hover {
      color: #216a38;
      background-color: white;
      transition: background-color 0.2s ease;
    }

    fa-icon {
      font-size: 2.5rem;
    }
  }

  &.large {
    a {
      font-size: 1.4rem;
      height: 60px;
      flex: 1 1 0;
      line-height: 60px;
      max-width: 450px;
      margin: 0 16px;

      text-overflow: ellipsis;
      white-space: nowrap;
      overflow: hidden;

      fa-icon {
        font-size: 1.5rem;
        line-height: 1.5rem;
      }

      @media #{$mat-md} {
        max-width: 350px;
        height: 50px;
        font-size: 1rem;
        line-height: 50px;

        fa-icon {
          font-size: 1.3rem;
          line-height: 1.5rem;
        }
      }
    }
  }
}
