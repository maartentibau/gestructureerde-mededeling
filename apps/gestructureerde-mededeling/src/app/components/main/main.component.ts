import { Breakpoints } from '@angular/cdk/layout';
import { Component } from '@angular/core';
import { Title } from '@angular/platform-browser';
import { FaIconLibrary } from '@fortawesome/angular-fontawesome';
import { faCheckCircle, faCog, faStar } from '@fortawesome/free-solid-svg-icons';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { DEFAULT_TITLE } from '../../core/core.constants';
import { ScreenService } from '../../core/services/screen.service';

@Component({
  selector: 'ogm-main',
  templateUrl: './main.component.html',
  styleUrls: ['./main.component.scss'],
})
export class MainComponent {
  readonly isSmallScreen$: Observable<boolean>;

  constructor(private screenService: ScreenService, private faIconLibrary: FaIconLibrary, private title: Title) {
    this.title.setTitle(DEFAULT_TITLE);

    this.isSmallScreen$ = this.screenService
      .observerBreakpoints()
      .pipe(
        map((breakpoints: { [key: string]: boolean }) =>
          Boolean(breakpoints[Breakpoints.XSmall] || breakpoints[Breakpoints.Small])
        )
      );

    this.faIconLibrary.addIcons(faCheckCircle, faCog, faStar);
  }
}
