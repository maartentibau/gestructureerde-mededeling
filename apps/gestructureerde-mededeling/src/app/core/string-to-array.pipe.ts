import { Pipe, PipeTransform } from '@angular/core';

@Pipe({
  name: 'stringToArray',
})
export class StringToArrayPipe implements PipeTransform {
  transform(value: string | null | undefined): string[] {
    if ((value && typeof value === 'string') || typeof value === 'number') {
      return Array.from(String(value));
    }

    return [];
  }
}
