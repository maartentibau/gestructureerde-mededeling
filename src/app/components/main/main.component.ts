import { Breakpoints } from '@angular/cdk/layout';
import { AsyncPipe } from '@angular/common';
import { Component, inject } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import { Title } from '@angular/platform-browser';
import { RouterLink } from '@angular/router';
import { FaIconLibrary, FontAwesomeModule } from '@fortawesome/angular-fontawesome';
import { faCheckCircle, faCog, faStar } from '@fortawesome/free-solid-svg-icons';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { DEFAULT_TITLE } from '../../core/core.constants';
import { ScreenService } from '../../core/services/screen.service';

@Component({
  standalone: true,
  imports: [AsyncPipe, MatButtonModule, FontAwesomeModule, RouterLink],
  selector: 'ogm-main',
  templateUrl: './main.component.html',
  styleUrls: ['./main.component.scss'],
})
export class MainComponent {
  #screenService: ScreenService = inject(ScreenService);
  #faIconLibrary: FaIconLibrary = inject(FaIconLibrary);
  #title: Title = inject(Title);

  readonly isSmallScreen$: Observable<boolean>;

  constructor() {
    this.#title.setTitle(DEFAULT_TITLE);

    this.isSmallScreen$ = this.#screenService
      .observerBreakpoints()
      .pipe(
        map((breakpoints: { [key: string]: boolean }) =>
          Boolean(breakpoints[Breakpoints.XSmall] || breakpoints[Breakpoints.Small]),
        ),
      );

    this.#faIconLibrary.addIcons(faCheckCircle, faCog, faStar);
  }
}
