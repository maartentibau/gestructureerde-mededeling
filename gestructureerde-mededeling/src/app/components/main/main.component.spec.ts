import { AsyncPipe, JsonPipe, NgIf } from '@angular/common';
import { Component, Input } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MatButtonModule } from '@angular/material/button';
import { RouterTestingModule } from '@angular/router/testing';
import { FaIconLibrary } from '@fortawesome/angular-fontawesome';
import { ScreenService } from '../../core/services/screen.service';

import { MainComponent } from './main.component';

@Component({
  standalone: true,
  imports: [JsonPipe],
  selector: 'fa-icon',
  template: ` <div>{{ icon | json }}</div> `,
})
class MockFaIconComponent {
  @Input() icon: string | string[] | undefined;
}

describe('MainComponent', () => {
  let component: MainComponent;
  let fixture: ComponentFixture<MainComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [MainComponent],
      providers: [
        {
          provide: ScreenService,
          useValue: { observerBreakpoints: () => ({ pipe: jest.fn() }) },
        },
        { provide: FaIconLibrary, useValue: { addIcons: jest.fn() } },
      ],
    })
      .overrideComponent(MainComponent, {
        set: {
          imports: [NgIf, AsyncPipe, RouterTestingModule, MockFaIconComponent, MatButtonModule],
        },
      })
      .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(MainComponent);
    component = fixture.componentInstance;
  });

  describe('create & render', () => {
    it('should create', () => {
      expect(component).toBeTruthy();
    });

    it('should render', () => {
      // act
      fixture.detectChanges();

      // check
      expect(fixture).toMatchSnapshot();
    });
  });
});
