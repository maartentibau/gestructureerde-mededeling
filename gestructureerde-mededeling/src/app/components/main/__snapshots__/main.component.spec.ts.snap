// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainComponent create & render should render 1`] = `
<ogm-main
  isSmallScreen$="undefined"
>
  <div
    class="shortcuts large"
  >
    <a
      aria-disabled="false"
      class="mdc-button mdc-button--raised mat-mdc-raised-button mat-primary mat-mdc-button-base"
      color="primary"
      disableripple=""
      href="/create"
      mat-raised-button=""
      mat-ripple-loader-class-name="mat-mdc-button-ripple"
      mat-ripple-loader-disabled=""
      mat-ripple-loader-uninitialized=""
      title="Maak uw eigen gestructureerde mededeling"
    >
      <span
        class="mat-mdc-button-persistent-ripple mdc-button__ripple"
      />
      <span
        class="mdc-button__label"
      >
        <fa-icon>
          <div>
            [
    "fas",
    "star"
  ]
          </div>
        </fa-icon>
        Zelf Mededeling Maken
      </span>
      <span
        class="mat-mdc-focus-indicator"
      />
      <span
        class="mat-mdc-button-touch-target"
      />
    </a>
    <a
      aria-disabled="false"
      class="mdc-button mdc-button--raised mat-mdc-raised-button mat-unthemed mat-mdc-button-base"
      disableripple=""
      href="/generate"
      mat-raised-button=""
      mat-ripple-loader-class-name="mat-mdc-button-ripple"
      mat-ripple-loader-disabled=""
      mat-ripple-loader-uninitialized=""
      title="Genereer een willekeurige gestructureerde mededeling"
    >
      <span
        class="mat-mdc-button-persistent-ripple mdc-button__ripple"
      />
      <span
        class="mdc-button__label"
      >
        <fa-icon>
          <div>
            [
    "fas",
    "cog"
  ]
          </div>
        </fa-icon>
        Genereer Mededeling
      </span>
      <span
        class="mat-mdc-focus-indicator"
      />
      <span
        class="mat-mdc-button-touch-target"
      />
    </a>
    <a
      aria-disabled="false"
      class="mdc-button mdc-button--raised mat-mdc-raised-button mat-unthemed mat-mdc-button-base"
      disableripple=""
      href="/validate"
      mat-raised-button=""
      mat-ripple-loader-class-name="mat-mdc-button-ripple"
      mat-ripple-loader-disabled=""
      mat-ripple-loader-uninitialized=""
      title="Controleer een gestructureerde mededeling"
    >
      <span
        class="mat-mdc-button-persistent-ripple mdc-button__ripple"
      />
      <span
        class="mdc-button__label"
      >
        <fa-icon>
          <div>
            [
    "fas",
    "check-circle"
  ]
          </div>
        </fa-icon>
        Controleer Mededeling
      </span>
      <span
        class="mat-mdc-focus-indicator"
      />
      <span
        class="mat-mdc-button-touch-target"
      />
    </a>
  </div>
</ogm-main>
`;
